File: src/main/webapp/src/classes/DocumentReport.ts
import RuleViolation from "./RuleViolation";

/**
 * Document report contains PDF's name, list of rule violations and error code
 * Error codes:
 * 		0	--	No Errors
 * 		-1	--	Empty File Name
 */
export default class DocumentReport {
	name: string;
	fileName: string;
	ruleViolations: RuleViolation[];
	errorCode: number;

	constructor(name: string, fileName: string, ruleViolations: RuleViolation[], errorCode: number) {
		this.name = name;
		this.fileName = fileName;
		this.ruleViolations = ruleViolations;
		this.errorCode = errorCode;
	}
}

File: src/main/kotlin/com/github/darderion/mundaneassignmentpolice/checker/DocumentReport.kt
package com.github.darderion.mundaneassignmentpolice.checker

class DocumentReport(val name: String, val ruleViolations: List<RuleViolation>, val errorCode: Int) {
	companion object {
		val emptyFileName = DocumentReport("EmptyFileName", listOf(), -1)
	}
}
